src/main/java/gov/cabinetoffice/api/security/ExceptionHandlerFilter.java

```diff
@@ -0,0 +1,39 @@
+package gov.cabinetoffice.api.security;
+
+import com.auth0.jwt.exceptions.MissingClaimException;
+import com.auth0.jwt.exceptions.SignatureVerificationException;
+import com.auth0.jwt.exceptions.TokenExpiredException;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.NonNull;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+public class ExceptionHandlerFilter extends OncePerRequestFilter {
+    @Override
+    protected void doFilterInternal(@NonNull HttpServletRequest request, @NonNull HttpServletResponse response, @NonNull FilterChain filterChain) throws ServletException, IOException {
+        try {
+            filterChain.doFilter(request, response);
+        }
+        catch (TokenExpiredException | SignatureVerificationException | MissingClaimException e) {
+            String message = switch (e.getClass().getSimpleName()) {
+                case "TokenExpiredException" -> "JWT is expired";
+                case "SignatureVerificationException" -> "JWT is not valid";
+                case "MissingClaimException" -> "JWT is missing expected properties";
+                default -> "Invalid JWT";
+            };
+
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+            response.getWriter().write(message);
+            response.getWriter().flush();
+        }
+        catch (RuntimeException e) {
```

Comment:
sorry I didn't spot this earlier but catching runtime exceptions in Java isn't really the done thing. If you want a catch-all case for any other exceptions that could occur I would personally just add `Exception` to the catch block above and let it run through the default part of the switch statement.